### PR TITLE
[0.5.10] Optimize FTS/MultiEntry index insertion.

### PR DESCRIPTION
**Pull request by berickson1**
Since we have a transaction, we can do the multi-index creation while we're storing the data

### CODE REVIEW

File: src/SqlProviderBase.ts [hunk 1]
```diff
@@ -576,79 +576,77 @@ class SqlStore implements NoSqlProvider.DbStore {
...
+        }
+
+        let promise = SyncTasks.all(queries);
         if (this._verbose) {
             promise = promise.finally(() => {
                 console.log('SqlStore (' + this._schema.name + ') put: (' + (Date.now() - startTime) + 'ms): Count: ' + items.length);
             });
         }
-        return promise;
+        return promise.then(_.noop);
```
Comment (dryganets) [inline src/SqlProviderBase.ts]: Out of curiosity:
Why not just
return promise;

File: src/SqlProviderBase.ts [hunk 2]
```diff
@@ -576,79 +576,77 @@ class SqlStore implements NoSqlProvider.DbStore {
...
+        if (_.some(this._schema.indexes, index => index.multiEntry || (index.fullText && this._supportsFTS3))) {
+            let err: Error;
+            _.each(items, item => {
+                let key: string;
+                const err = _.attempt(() => {
+                    key = NoSqlProviderUtils.getSerializedKeyForKeypath(item, this._schema.primaryKeyPath);
+                });
+                if (err) {
+                    // Return early
+                    return false;
```
Comment (dryganets) [inline src/SqlProviderBase.ts]: Don't we want to push Promise.Rejected in this case?

File: src/SqlProviderBase.ts [hunk 3]
```diff
@@ -576,79 +576,77 @@ class SqlStore implements NoSqlProvider.DbStore {
...
+        // Also prepare mulltiEntry and FullText indexes
+        if (_.some(this._schema.indexes, index => index.multiEntry || (index.fullText && this._supportsFTS3))) {
+            let err: Error;
+            _.each(items, item => {
+                let key: string;
+                const err = _.attempt(() => {
+                    key = NoSqlProviderUtils.getSerializedKeyForKeypath(item, this._schema.primaryKeyPath);
+                });
+                if (err) {
+                    // Return early
```
Comment (dryganets) [inline src/SqlProviderBase.ts]: Don't we need to push rejected promise here to queries?

File: src/SqlProviderBase.ts (hunk 1 above)
Comment (berickson1) [inline src/SqlProviderBase.ts]: SyncTasks.all typings will return a promise with array return value.  _.noop changes return type back to undefined

**Pull request merged**